readline_deps are correctly detected, but not used during the compilation
resulting in ld symbol referencing errors.

This was offered and accepted upstream:
https://gitlab.gnome.org/GNOME/gjs/-/merge_requests/543

--- gjs-1.66.1/meson.build
+++ gjs-1.66.1/meson.build
@@ -474,6 +474,10 @@ if build_cairo
     endif
 endif
 
+if build_readline
+    libgjs_dependencies += readline_deps
+endif
+
 if build_profiler
     libgjs_dependencies += profiler_deps
 endif
